scapy/contrib/isotp.py

```diff
@@ -168,6 +171,120 @@ def defragment(can_frames, use_extended_addressing=None):
         return results[0]
 
 
+class ISOTPHeader(CAN):
+    name = 'ISOTPHeader'
+    fields_desc = [
+        FlagsField('flags', 0, 3, ['error', 'remote_transmission_request', 'extended']),  # noqa: E501
+        XBitField('identifier', 0, 29),
+        FieldLenField('length', None, fmt="B", length_of="reserved",
```

Comment:
This won’t do anything, as it will be called before the payload building. You can use a simple `ByteField("length", None)` and the trick in `post_dissect`